src/main/java/org/shanerx/tradeshop/commands/commandrunners/ShopUserCommand.java

```diff
@@ -0,0 +1,260 @@
+                            break eachOwnedShop;
+                        }
+                        break;
+                }
+
+                PlayerShopChangeEvent changeEvent = new PlayerShopChangeEvent(pSender, shop, change, new ObjectHolder<OfflinePlayer>(target));
+                Bukkit.getPluginManager().callEvent(changeEvent);
+                if (changeEvent.isCancelled()) return;
+
+                boolean success = false;
+
+                switch (change) {
+                    case SET_MANAGER:
+                    case SET_MEMBER:
+                        success = shop.setUser(target.getUniqueId(), role);
+                        break;
+                    case ADD_MEMBER:
+                    case ADD_MANAGER:
+                        success = shop.addUser(target.getUniqueId(), role);
+                        break;
+                    case REMOVE_USER:
+                        success = shop.removeUser(target.getUniqueId());
+                        break;
+                }
+
+                if (success)
+                    updateStatuses.put(shop.getShopLocationAsSL().serialize(), UserOperationStatus.SUCCESSFUL.toString());
+                else
+                    updateStatuses.put(shop.getShopLocationAsSL().serialize(), UserOperationStatus.FAILED.toString());
+            }
+        }
+
+        Message.UPDATED_SHOP_USERS.sendUserEditMultiLineMessage(pSender, Collections.singletonMap(Variable.UPDATED_SHOPS, updateStatuses));
+    }
+
+
+    //region Util Methods
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Checks if targeted player exists and if player is looking at a shop while not targetting all owned shops
+     *
+     * @param applyAllOwned Is Player targetting all owned shops
+     * @return shop if found or null if not needed; returning null while setting target to null indicates failure, command should respond with an immediate blank return.
+     */
+    private Shop shopUserCommandStart(boolean applyAllOwned) {
+        target = Bukkit.getOfflinePlayer(command.getArgAt(1));
+        if (!target.hasPlayedBefore()) {
+            Message.PLAYER_NOT_FOUND.sendMessage(pSender);
+            target = null;
+            return null;
+        }
+
+        if (!applyAllOwned) {
+            Shop shop = findShop();
+
+            if (shop == null) {
+                // Message.NO_SIGHTED_SHOP.sendMessage(pSender); // Message is sent by findShop()
+                target = null;
+                return null;
+            }
+
+            if (!shop.getOwner().getUUID().equals(pSender.getUniqueId())
+                    || (Setting.UNLIMITED_ADMIN.getBoolean() && Permissions.isAdminEnabled(pSender))) {
+                Message.NO_SHOP_PERMISSION.sendMessage(pSender);
+                target = null;
+                return null;
+            }
+
+            return shop;
+        }
+
+        return null;
+    }
```

Comment:
yes I think you are right, was tripping there. Still, this should not indicate failure by setting global variables to null. Also I still think it is safer to make target local to prevent accidental modification in the future, and move to exception based failure.

Reply:
target gets passed to the method now.